google/cloud/storage/async/client.cc

```diff
@@ -38,6 +38,53 @@ AsyncClient::AsyncClient(Options options) {
 AsyncClient::AsyncClient(std::shared_ptr<AsyncConnection> connection)
     : connection_(std::move(connection)) {}
 
+future<StatusOr<std::pair<AsyncReader, AsyncToken>>> AsyncClient::ReadObject(
+    BucketName const& bucket_name, std::string object_name, Options opts) {
+  auto request = google::storage::v2::ReadObjectRequest{};
+  request.set_bucket(bucket_name.FullName());
+  request.set_object(std::move(object_name));
+  return ReadObject(std::move(request), std::move(opts));
+}
+
+future<StatusOr<std::pair<AsyncReader, AsyncToken>>> AsyncClient::ReadObject(
+    google::storage::v2::ReadObjectRequest request, Options opts) {
+  return connection_
+      ->ReadObject(
+          {std::move(request),
+           internal::MergeOptions(std::move(opts), connection_->options())})
+      .then([](auto f) -> StatusOr<std::pair<AsyncReader, AsyncToken>> {
+        auto impl = f.get();
+        if (!impl) return std::move(impl).status();
+        auto t = storage_internal::MakeAsyncToken(impl->get());
+        return std::make_pair(AsyncReader(*std::move(impl)), std::move(t));
+      });
+}
+
+future<StatusOr<ReadPayload>> AsyncClient::ReadObjectRange(
+    BucketName const& bucket_name, std::string object_name, std::int64_t offset,
+    std::int64_t limit, Options opts) {
+  auto request = google::storage::v2::ReadObjectRequest{};
+  request.set_bucket(bucket_name.FullName());
+  request.set_object(std::move(object_name));
+  request.set_read_offset(offset);
+  request.set_read_limit(limit);
+
+  return connection_->ReadObjectRange(
+      {std::move(request),
+       internal::MergeOptions(std::move(opts), connection_->options())});
```

Comment:
Could not this forward to `ReadObjectRange(request, ...)`, to match the layout of the `ReadObject()` overloads?

Reply:
Done.